### PR TITLE
LPS-182671 Clean up subclasses of class BaseModelPermissionChecker | part 1

### PR DESCRIPTION
**Pull request by W2jx**
I'm eventually going to clean up the method references in class MBDiscussionPermission, as requested in the [comment](https://github.com/liferay-core-infra/liferay-portal/pull/4411#issuecomment-1533572068), I start by removing the deprecated classes.

Previous: https://github.com/liferay-core-infra/liferay-portal/pull/4411

### CODE REVIEW

**Comment by Kvale1733** [comment]
ci:test:document

**Comment by Kvale1733** [comment]
ci:test:message-boards

**Comment by Kvale1733** [comment]
ci:test:stable

**Comment by W2jx** [comment]
no conflicts(5/11 10:11)

**Comment by zhangyan0701** [comment]
No regressions found.

**Comment by W2jx** [comment]
no conflicts(5/12 10:36)

**Comment by W2jx** [comment]
no conflicts(5/15 9:43)

File: modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/internal/security/permission/resource/MBThreadModelResourcePermission.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+@Component(
+	property = "model.class.name=com.liferay.message.boards.model.MBThread",
+	service = ModelResourcePermission.class
+)
+public class MBThreadModelResourcePermission
+	implements ModelResourcePermission<MBThread> {
+
+	@Override
+	public void check(
+			PermissionChecker permissionChecker, long messageId,
```
Comment (dantewang) [inline modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/internal/security/permission/resource/MBThreadModelResourcePermission.java:40]: @W2jx The `ModelResourcePermission<T>` interface names the `long` parameter `primaryKey`, we should assume that the `primaryKey` is the pk of the `T`.

Do you see this being a common pattern in portal? If not, please change it to `threadId` and use it as `threadId`; you can always get the root message id from the thread id though.

Comment (dantewang) [reply modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/internal/security/permission/resource/MBThreadModelResourcePermission.java:40]: In this commit message, you say `sup-class`. It's confusing. I thought you meant `sub-class` and made a typo. Please always say the full word, "super class".

**Review by dantewang — CHANGES_REQUESTED** [review]

**Pull request closed without merging**